### PR TITLE
Refactor Python tests.

### PR DESCRIPTION
**Pull request by trivialfis**
- Remove nose tests.  Now only pytest is required, with the benefit of reducing Travis test time.
- Format all python test code.  I don't know why flake8 on test farm doesn't report any error.  But it keeps bugging me on my editor.  If it's too much noise for review I will change it back.
- Display fulltrace when tests fail, should be helpful for #3720.

### CODE REVIEW

**Comment by trivialfis** [comment]
@hcho3 Jenkins and docker configurations are changed, might need some help. :)

**Comment by hcho3** [comment]
Is pytest faster than nose?

File: tests/ci_build/test_mgpu.sh [hunk 1]
```diff
@@ -4,5 +4,5 @@ set -e
 cd python-package
 python setup.py install --user
 cd ..
-python -m nose -v --eval-attr='(not slow) and mgpu' tests/python-gpu/
+pytest -s --fulltrace -m "not slow" tests/python-gpu
```
Comment (hcho3) [inline tests/ci_build/test_mgpu.sh]:
```suggestion
pytest -v -s --fulltrace -m "(not slow) and mgpu" tests/python-gpu
```
1. Add `-v` to print out name of all tests
2. Add `and mgpu` to preserve existing behavior

File: tests/ci_build/test_gpu.sh [hunk 2]
```diff
@@ -4,6 +4,5 @@ set -e
 cd python-package
 python setup.py install --user
 cd ..
-python -m nose -v --eval-attr='(not slow) and (not mgpu)' tests/python-gpu/
+pytest -s --fulltrace -m "(not mgpu) and (not slow)" tests/python-gpu
```
Comment (hcho3) [inline tests/ci_build/test_gpu.sh]:
```suggestion
pytest -v -s --fulltrace -m "(not mgpu) and (not slow)" tests/python-gpu
```

File: Jenkinsfile [hunk 3]
```diff
@@ -96,11 +96,11 @@ def buildPlatformCmake(buildName, conf, nodeReq, dockerTarget) {
                 # Test the wheel for compatibility on a barebones CPU container
                 ${dockerRun} release ${dockerArgs} bash -c " \
                     pip install --user python-package/dist/xgboost-*-none-any.whl && \
-                    python -m nose -v tests/python"
+		    pytest -v --fulltrace -s tests/python
                 # Test the wheel for compatibility on CUDA 10.0 container
                 ${dockerRun} gpu --build-arg CUDA_VERSION=10.0 bash -c " \
                     pip install --user python-package/dist/xgboost-*-none-any.whl && \
-                    python -m nose -v --eval-attr='(not slow) and (not mgpu)' tests/python-gpu"
+		    pytest -m -s --fulltrace "(not mgpu) and (not slow)" tests/python-gpu
```
Comment (hcho3) [inline Jenkinsfile]:
```suggestion
                    pytest -v -s --fulltrace -m '(not mgpu) and (not slow)' tests/python-gpu"
```

This should fix the broken Jenkins test.

**Comment by hcho3** [comment]
@trivialfis I left some comments, let me know what you think

**Comment by trivialfis** [comment]
@hcho3 Thanks!
No, pytest is not faster, it's just somehow Travis has been running both nosetest and pytest.

File: Jenkinsfile [hunk 4]
```diff
@@ -96,11 +96,11 @@ def buildPlatformCmake(buildName, conf, nodeReq, dockerTarget) {
                 # Test the wheel for compatibility on a barebones CPU container
                 ${dockerRun} release ${dockerArgs} bash -c " \
                     pip install --user python-package/dist/xgboost-*-none-any.whl && \
-                    python -m nose -v tests/python"
+		    pytest -v --fulltrace -s tests/python
                 # Test the wheel for compatibility on CUDA 10.0 container
                 ${dockerRun} gpu --build-arg CUDA_VERSION=10.0 bash -c " \
                     pip install --user python-package/dist/xgboost-*-none-any.whl && \
-                    python -m nose -v --eval-attr='(not slow) and (not mgpu)' tests/python-gpu"
+		    pytest -v -s --fulltrace -m "(not mgpu) and (not slow)" tests/python-gpu
```
Comment (hcho3) [inline Jenkinsfile]:
```suggestion
		    pytest -v -s --fulltrace -m '(not mgpu) and (not slow)' tests/python-gpu"
```
We need to close the double quote (see two lines up). Also, we should use single quotes inside double quotes. If you don't mind, I can apply the fix

**Comment by hcho3** [comment]
@trivialfis Got it.

**Pull request merged**